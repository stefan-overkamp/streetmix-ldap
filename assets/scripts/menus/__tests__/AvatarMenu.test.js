/* eslint-env jest */
import React from 'react'
import { screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { renderWithRedux } from '../../../../test/helpers/render'
import AvatarMenu from '../AvatarMenu'

const user = {
  id: 'foo'
}

describe('AvatarMenu', () => {
  it('renders user avatar', () => {
    const { asFragment } = renderWithRedux(<AvatarMenu user={user} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders user avatar for admin', () => {
    const user = {
      id: 'foo',
      roles: ['ADMIN']
    }
    const { asFragment } = renderWithRedux(<AvatarMenu user={user} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('calls click handler', () => {
    const onClick = jest.fn()
    renderWithRedux(<AvatarMenu user={user} onClick={onClick} />)
    userEvent.click(screen.getByText(user.id))
    expect(onClick).toHaveBeenCalled()
  })
})
