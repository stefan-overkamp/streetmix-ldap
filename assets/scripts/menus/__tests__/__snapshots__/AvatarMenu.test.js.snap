// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarMenu renders user avatar 1`] = `
<DocumentFragment>
  <button
    class="menu-attached menu-avatar"
  >
    <span
      class="avatar"
    >
      <img
        alt="foo"
      />
    </span>
    <span
      class="user-id"
    >
      foo
    </span>
  </button>
</DocumentFragment>
`;

exports[`AvatarMenu renders user avatar for admin 1`] = `
<DocumentFragment>
  <button
    class="menu-attached menu-avatar"
  >
    <span
      class="avatar"
    >
      <img
        alt="foo"
      />
    </span>
    <svg
      class="svg-inline--fa menu-avatar-admin fa-crown"
    />
    <span
      class="user-id"
    >
      foo
    </span>
  </button>
</DocumentFragment>
`;
